### ISSUE TITLE
Runtime error  CUDA out of memory

### ISSUE THREAD

**Issue by IUUI11** [znxlwm/UGATIT-pytorch#75]
![image](https://user-images.githubusercontent.com/41223176/202903190-ec01dc14-88b4-49d7-807a-2f715394d897.png)
**I was trying to run this model on 2080ti，but it always said I do not have enough GPU memory**.
There is no other process using GPU
![image](https://user-images.githubusercontent.com/41223176/202903281-f7c3ce4c-5a24-4448-a8a3-8461b41c675e.png)

**Comment by kirill-ionkin** [comment]
Hello, are you sure, you check nvidia-smi, than the procces still working?
I'm checking and rewriting this repo few time and this repo unfortunately has a little bit small mistackes, like CUDA out of memory when evaluating model, because there isn't with torch.no_grad().

Also, be sure, you use light version of model, only batch_size=1 and other paramaters, which are affect to the model architecture - try to decrease them until your model can stored on GPU device.

**Comment by vasyza** [comment]
Same problem

**Comment by gogobd** [comment]
Same problem, even with "--light=True" memory is running out after 1k steps... NVIDIA GeForce RTX 3060, 12GB GPU Memory...

**Comment by kirill-ionkin** [comment]
> Same problem, even with "--light=True" memory is running out after 1k steps... NVIDIA GeForce RTX 3060, 12GB GPU Memory...

On my opinion, there is better solution to unpaired transfer style, like VSAIT: Unpaired Image Translation via Vector Symbolic Architectures for example.

**Comment by gogobd** [comment]
> > Same problem, even with "--light=True" memory is running out after 1k steps... NVIDIA GeForce RTX 3060, 12GB GPU Memory...
> 
> On my opinion, there is better solution to unpaired transfer style, like VSAIT: Unpaired Image Translation via Vector Symbolic Architectures for example.

Thanks @kirill-ionkin for pointing it out, I was trying VSAIT recently; for reference I found it here: <https://github.com/facebookresearch/vsait>

**Comment by BigRootCaptain** [comment]
Yeah,and I've noticed that after a certain number of iterations, the GPU memory requirement grows. I'm training with single-channel images, batch_size is 2, and about 1000 pairs of images. And, it ends up needing around 20GB of VRAM,so big.
